api/groups/blockGroup.go

```diff
@@ -143,6 +159,66 @@ func (bg *blockGroup) getBlockByRound(c *gin.Context) {
 	shared.RespondWith(c, http.StatusOK, gin.H{"block": block}, "", shared.ReturnCodeSuccess)
 }
 
+func (bg *blockGroup) getAlteredAccountsByNonce(c *gin.Context) {
+	nonce, err := getQueryParamNonce(c)
+	if err != nil {
+		shared.RespondWithValidationError(c, errors.ErrGetAlteredAccountsForBlock, errors.ErrInvalidBlockRound)
+		return
+	}
+
+	tokensFilter, withMetadata, err := parseAlteredAccountsForBlockQueryOptions(c)
+	if err != nil {
+		shared.RespondWithValidationError(c, errors.ErrGetAlteredAccountsForBlock, err)
+	}
+
+	options := api.GetAlteredAccountsForBlockOptions{
+		GetBlockParameters: api.GetBlockParameters{
+			RequestType: api.BlockFetchTypeByNonce,
+			Nonce:       nonce,
+		},
+		TokensFilter: tokensFilter,
+		WithMetadata: withMetadata,
+	}
+
+	start := time.Now()
+	alteredAccountsResponse, err := bg.getFacade().GetAlteredAccountsForBlock(options)
+	logging.LogAPIActionDurationIfNeeded(start, "API call: GetAlteredAccountsForBlock by nonce")
+	if err != nil {
+		shared.RespondWithInternalError(c, errors.ErrGetAlteredAccountsForBlock, err)
+		return
+	}
+
+	shared.RespondWith(c, http.StatusOK, gin.H{"accounts": alteredAccountsResponse}, "", shared.ReturnCodeSuccess)
```

Comment:
Not sure, I think we have an utility function for `RespondWithSuccess` - but can be left as it is.

Reply:
indeed. used